### PR TITLE
Fix SudoContractProposal and ExecuteContractProposal

### PR DESCRIPTION
**Pull request by the-frey**
Follow up fix for #730

Add missing `SudoContractProposal` and `ExecuteContractProposal` where it was missinng in types/codec and basic validation unit test.

Could be some more to this that I've missed.

### CODE REVIEW

**Comment by the-frey** [comment]
Will also be running some manual tests tomorrow to check if this resolves the issues we've seen with gov/sudo.

**Comment by the-frey** [comment]
Update: we have run this on a testnet with a sudo contract and successfully moved funds via a gov prop. Our auto-CI hook that was failing with 0.25.0 also passes when we substitute in this branch, so we're pretty confident it's the fix.

**Review by maurolacy — APPROVED** [review]
LGTM. Perhaps @alpe wants to take a look.

File: x/wasm/types/codec.go [hunk 1]
```diff
@@ -44,6 +46,8 @@ func RegisterInterfaces(registry types.InterfaceRegistry) {
 		&StoreCodeProposal{},
 		&InstantiateContractProposal{},
 		&MigrateContractProposal{},
+		&SudoContractProposal{},
+		&ExecuteContractProposal{},
```
Comment (pinosu) [inline x/wasm/types/codec.go:50]: 👍

File: x/wasm/types/proposal_test.go [hunk 2]
```diff
@@ -317,6 +317,61 @@ func TestValidateMigrateContractProposal(t *testing.T) {
 	}
 }
 
+func TestValidateSudoContractProposal(t *testing.T) {
```
Comment (pinosu) [inline x/wasm/types/proposal_test.go:320]: nice test!

File: x/wasm/types/proposal_test.go [hunk 3]
```diff
@@ -317,6 +317,61 @@ func TestValidateMigrateContractProposal(t *testing.T) {
...
+			err := spec.src.ValidateBasic()
+			if spec.expErr {
+				require.Error(t, err)
+			} else {
+				require.NoError(t, err)
+			}
+		})
+	}
+}
+
```
Comment (pinosu) [inline x/wasm/types/proposal_test.go:374]: Could you add some unit tests also for **ExecuteContractProposal** ?

File: x/wasm/types/test_fixtures.go [hunk 4]
```diff
@@ -257,6 +257,25 @@ func MigrateContractProposalFixture(mutators ...func(p *MigrateContractProposal)
 	return p
 }
 
+func SudoContractProposalFixture(mutators ...func(p *SudoContractProposal)) *SudoContractProposal {
+	const (
+		contractAddr = "cosmos14hj2tavq8fpesdwxxcu44rty3hh90vhujrvcmstl4zr3txmfvw9s4hmalr"
+		anyAddress   = "cosmos1qyqszqgpqyqszqgpqyqszqgpqyqszqgpqyqszqgpqyqszqgpqyqs2m6sx4"
```
Comment (pinosu) [inline x/wasm/types/test_fixtures.go:263]: unused

**Review by pinosu — COMMENTED** [review]
lgtm. 👍  
I added just some minor comments.
Also it would be nice if you could update the description mentioning that also ExecuteContractProposal was added.

File: x/wasm/types/test_fixtures.go (hunk 4 above)
Comment (the-frey) [reply x/wasm/types/test_fixtures.go:263]: SudoContractProposalFixture is used in the test above?

File: x/wasm/types/proposal_test.go (hunk 3 above)
Comment (the-frey) [reply x/wasm/types/proposal_test.go:374]: Added a unit test for it 👍

**Comment by the-frey** [comment]
@pinosu have updated the description and some additional test cases as you suggested.

**Review by webmaster128 — APPROVED** [review]

**Comment by ethanfrey** [comment]
Good find. I have a mental block with amino. Can't believe we **still** need to register all this stuff.

**Pull request merged**